python/lsst/faro/base/MatchedCatalogBase.py

```diff
@@ -11,81 +12,117 @@ class MatchedBaseTaskConnections(pipeBase.PipelineTaskConnections,
+            self.inputs.remove("externalSkyWcsTractCatalog")
+            self.inputs.remove("externalSkyWcsGlobalCatalog")
+        if config.doApplyExternalPhotoCalib:
+            if config.useGlobalExternalPhotoCalib:
+                self.inputs.remove("externalPhotoCalibTractCatalog")
+            else:
+                self.inputs.remove("externalPhotoCalibGlobalCatalog")
+        else:
+            self.inputs.remove("externalPhotoCalibTractCatalog")
+            self.inputs.remove("externalPhotoCalibGlobalCatalog")
 
 
 class MatchedBaseTaskConfig(pipeBase.PipelineTaskConfig,
                             pipelineConnections=MatchedBaseTaskConnections):
     match_radius = pexConfig.Field(doc="Match radius in arcseconds.", dtype=float, default=1)
-    apply_external_wcs = pexConfig.Field(doc="Apply correction to coordinates with e.g. a jointcal WCS.",
-                                         dtype=bool, default=False)
-    wcsDimensions = pexConfig.ListField(doc="Override the dimensions of the astrometric calibration objects",
-                                        dtype=str,
-                                        default=MatchedBaseTaskConnections.astrom_calibs.dimensions)
+    doApplyExternalSkyWcs = pexConfig.Field(doc="Whether or not to use the external wcs.",
+                                            dtype=bool, default=False)
+    useGlobalExternalSkyWcs = pexConfig.Field(doc="Whether or not to use the global external wcs.",
+                                              dtype=bool, default=False)
+    doApplyExternalPhotoCalib = pexConfig.Field(doc="Whether or not to use the external photoCalib.",
+                                                dtype=bool, default=False)
+    useGlobalExternalPhotoCalib = pexConfig.Field(doc="Whether or not to use the global external photoCalib.",
+                                                  dtype=bool, default=False)
 
 
 class MatchedBaseTask(pipeBase.PipelineTask):
 
     ConfigClass = MatchedBaseTaskConfig
     _DefaultName = "matchedBaseTask"
 
-    def __init__(self, config: pipeBase.PipelineTaskConfig, *args, **kwargs):
+    def __init__(self, config: MatchedBaseTaskConfig, *args, **kwargs):
         super().__init__(*args, config=config, **kwargs)
         self.radius = self.config.match_radius
         self.level = "patch"
 
-    def run(self, source_catalogs, photo_calibs, astrom_calibs, vIds, wcs, box, apply_external_wcs):
+    def run(self, sourceCatalogs, photoCalibs, astromCalibs, dataIds, wcs, box,
+            doApplyExternalSkyWcs=False, doApplyExternalPhotoCalib=False):
         self.log.info("Running catalog matching")
         radius = geom.Angle(self.radius, geom.arcseconds)
-        srcvis, matched = match_catalogs(source_catalogs, photo_calibs, astrom_calibs, vIds, radius,
-                                         apply_external_wcs, logger=self.log)
+        srcvis, matched = match_catalogs(sourceCatalogs, photoCalibs, astromCalibs, dataIds, radius,
```

Comment:
As long as we are making change to camelCase for consistency, it seems that this function should be matchCatalogs